core/src/test/scala/JwtSpec.scala

```diff
@@ -266,7 +266,13 @@ class JwtSpec extends UnitSpec with Fixture {
     it("should invalidate wrong algos") {
       val token = "eyJ0eXAiOiJKV1QiLCJhbGciOiJXVEYifQ.e30"
       assert(Jwt.decode(token).isFailure)
-      intercept[JwtNonSupportedAlgorithm] { Jwt.decode(token).get }
+      intercept[JwtNonEmptyAlgorithmException] { Jwt.decode(token).get }
```

Comment:
I would try to keep the original exception if possible. There is an algorithm after all.